exp/api/v1beta1/azuremachinepool_webhook_test.go

```diff
@@ -260,13 +260,15 @@ func TestAzureMachinePool_Default(t *testing.T) {
 	g.Expect(publicKeyNotExistTest.amp.Spec.Template.SSHPublicKey).NotTo(BeEmpty())
 }
 
-func createMachinePoolWithtMarketPlaceImage(publisher, offer, sku, version string, terminateNotificationTimeout *int) *AzureMachinePool {
+func createMachinePoolWithMarketPlaceImage(publisher, offer, sku, version string, terminateNotificationTimeout *int) *AzureMachinePool {
 	image := infrav1.Image{
 		Marketplace: &infrav1.AzureMarketplaceImage{
-			Publisher: publisher,
-			Offer:     offer,
-			SKU:       sku,
-			Version:   version,
+			ImagePlan: infrav1.ImagePlan{
```

Comment:
just to confirm,  is this a breaking change to the `AzureMarketPlaceImage` CRD?

Reply:
From programmatic perspective yes, from YAML it should remain the same as far as I'm aware, as the fields are inlined.